### PR TITLE
Adds support for outputting HTML

### PR DESCRIPTION
**Pull request by lukecampbell**
- Added two new command line arguments:
  - `-o --output` 
  - `-f --format` 

Note: I haven't implemented the REST of the functionality for the output, in other words, you can't use -o anywhere other than for HTML output.

### CODE REVIEW

**Comment by lukecampbell** [comment]
@daf

**Comment by lukecampbell** [comment]
screen shot 2015-07-10 at 1 34 08 pm

File: requirements.txt [hunk 1]
```diff
@@ -5,3 +5,4 @@ lxml>=3.2.1
 cf_units==0.1.0
 requests>=2.2.1
 python-dateutil>=2.2
+Jinja2==2.7.3
```
Comment (daf) [inline requirements.txt]: y u exact `==` a version

Comment (lukecampbell) [inline requirements.txt]: Oh, that was because I ran this
```
pip freeze | grep -i jinja >> requirements.txt
```

I'll pop out the `==`

**Comment by lukecampbell** [comment]
screen shot 2015-07-10 at 2 23 20 pm

**Comment by daf** [comment]
Looks great.  Minor comment about the argument handling, it doesn't quite feel UNIX-y.  `-f` should probably be `text|html` (not `stdout`), and `-o` should accept `-` as UNIX convention to stdout - that way I can pipe the HTML to something if I want.  I also envision more output types (json anyone?).  This would decouple formatting from output instead of those if blocks.

Also, text output could be made a jinja template (right?)

**Comment by lukecampbell** [comment]
Wait one sec, I got a minor refactor that will make it more flexible

**Comment by lukecampbell** [comment]
ok now you can do something like
```
cchecker.py -t acdd -f html dataset.nc
```
And it will spit out a bunch of HTML text

Might be handy for a micro web service ** hint hint **

**Comment by daf** [comment]
- glyphs should toggle collapse when clicked (not just heading)
- priority field is redundant

File: compliance_checker/runner.py [hunk 2]
```diff
@@ -25,7 +26,7 @@ class ComplianceChecker(object):
     the command line or can be used via import.
     """
     @classmethod
-    def run_checker(cls, ds_loc, checker_names, verbose, criteria):
+    def run_checker(cls, ds_loc, checker_names, verbose, criteria, output_filename='stdout', output_format='stdout'):
```
Comment (daf) [inline compliance_checker/runner.py]: should these defaults be changed to `'-'` and `'text'` respectively?

**Comment by daf** [comment]
Corrective action table only appears to have top level corrective items.  For instance, this text output:

```
Name                             Priority:     Score:Reasoning
--------------------------------------------------------------------------------
conventions                            :3:     0/ 1 : Conventions field is not
                                                      present
std_name                               :3:     0/ 2 :  
    u                                  :3:     0/ 1 :  
        legal                          :3:     0/ 1 : The standard name
                                                      'eastward_surface_velocity
                                                      ' is not in standard name
                                                      table
    v                                  :3:     0/ 1 :  
        legal                          :3:     0/ 1 : The standard name
                                                      'northward_surface_velocit
                                                      y' is not in standard name
                                                      table
time                                   :3:     2/ 4 :  
    time                               :3:     2/ 4 :  
        has_calendar                   :1:     0/ 1 :  
        valid_calendar                 :1:     0/ 1 :  
```

Only shows this in HTML:

**Comment by lukecampbell** [comment]
It should be everything that has a message.

**Comment by lukecampbell** [comment]
What dataset are you using?

**Comment by lukecampbell** [comment]
Oh, I didn't know children results could have even more children.

**Comment by daf** [comment]
Yeah, a complexity that needs to be removed at some point.  Dataset is https://s3.amazonaws.com/paegan/resources/marcooshfradar20120331.nc just in case

**Comment by lukecampbell** [comment]
screen shot 2015-07-10 at 3 08 12 pm

**Comment by daf** [comment]
Looking good!  I have some minor ideas of where to take this but none that require a hold up here.

**Comment by abirger** [comment]
@daf , it looks good already! While perfecting it, just don't forget about the perfect being the enemy of you-know-what...

**Comment by kwilcox** [comment]
this is awesome!

**Pull request merged**